testHelpers/index.js

```diff
@@ -59,6 +59,19 @@ const saslConnectionOpts = () =>
     },
   })
 
+let callCount = 0
```

Comment:
This callcount is now shared with every test that imports this module. It should at least be scoped, i.e:

```javascript
const saslConnectionDynamicOpts = () => {
  let callCount = 0

  const baseSaslConnectionOpts = saslConnectionOpts()
  const sasl = async () => Object.assign(
    {},
    baseSaslConnectionOpts.sasl,
    callCount: ++callCount
  )

  return Object.assign({}, baseSaslConnectionOpts, { sasl })
}
```